### PR TITLE
Remove dyn buf from p2p reader

### PR DESCRIPTION
**Pull request by vlad9486**

### CODE REVIEW

**Review by bkontur — CHANGES_REQUESTED** [review]
@vlad9486 
Vlad, please check a lots of older clippy stuff in this PR,
and rebase to develop also

File: tezos/encoding/src/binary_reader.rs [hunk 1]
```diff
@@ -102,22 +102,51 @@ macro_rules! safe {
...
+    let temp = *buf;
+    let mut buf_slice = safe!(buf, upper, {
+        let (slice, remaining) = buf.split_at(upper);
+        *buf = remaining;
+        slice
+    });
+    let res = f(&mut buf_slice);
+    if !buf_slice.is_empty() {
+        let initial_length = temp.len();
+        let new_length = buf_slice.len() + (*buf).len();
```
[the remark this answers is not in the document]
Comment (bkontur) [reply tezos/encoding/src/binary_reader.rs:142]: @vlad9486 
please, Vlad

**Review by bkontur — CHANGES_REQUESTED** [review]
@vlad9486 @akoptelov 
I guess, there is just one commetnt with that arithemtics still unresolved

File: tezos/encoding/src/binary_reader.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlad9486) [reply tezos/encoding/src/binary_reader.rs]: Sorry for late respond. I updated the code. I see no need to do checked arithmetic and `safe!`. Refactored my code to make it obvious.

**Comment by bkontur** [comment]
@vlad9486 
please, check this panics and add Err handling instead:

https://github.com/simplestaking/tezedge/pull/501/files#diff-88ff5d321cc7a89f7e1c9ceaeff8e1971bfcf02315ebc32c73817f4e797caf5fR1066

https://github.com/simplestaking/tezedge/pull/501/files#diff-88ff5d321cc7a89f7e1c9ceaeff8e1971bfcf02315ebc32c73817f4e797caf5fR1094

https://github.com/simplestaking/tezedge/pull/501/files#diff-88ff5d321cc7a89f7e1c9ceaeff8e1971bfcf02315ebc32c73817f4e797caf5fR1104

**Review by bkontur — CHANGES_REQUESTED** [review]
@vlad9486 
please, check these panics and add Err handling instead:

https://github.com/simplestaking/tezedge/pull/501/files#diff-88ff5d321cc7a89f7e1c9ceaeff8e1971bfcf02315ebc32c73817f4e797caf5fR1066

https://github.com/simplestaking/tezedge/pull/501/files#diff-88ff5d321cc7a89f7e1c9ceaeff8e1971bfcf02315ebc32c73817f4e797caf5fR1094

https://github.com/simplestaking/tezedge/pull/501/files#diff-88ff5d321cc7a89f7e1c9ceaeff8e1971bfcf02315ebc32c73817f4e797caf5fR1104

**Comment by vlad9486** [comment]
@bkontur There are tests. Panic in test is ok.

**Comment by bkontur** [comment]
> 
> 
> @bkontur There are tests. Panic in test is ok.

ok, right :), I didnt realize it it in test,

@akoptelov 
Alex, please, could you also do a quick check?

**Comment by vlad9486** [comment]
@bkontur We can do this
```
        encode_value(&mut data, &value, &encoding)
            .expect_err("Encoding an empty enum value should not succeed");
```
Clippy will be happy (I hope), and code is more clear

**Comment by bkontur** [comment]
@vlad9486 
Vlad could you please rebase and squash this?

**Comment by akoptelov** [comment]
@bkontur @vlad9486 Does it make much sense in doing this now, and then trowing away as nom is fully adopted?

**Comment by bkontur** [comment]
> 
> 
> @bkontur @vlad9486 Does it make much sense in doing this now, and then trowing away as nom is fully adopted?

I dont know :), you tell me :D,
if this will go away and it does not fix anyhting critical, we can close it

**Comment by vlad9486** [comment]
@bkontur yes, let's close it.

**Pull request closed without merging**